platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/IdpMetadataTest.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.client;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNot.not;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+public class IdpMetadataTest {
+  public static final String CACHE_DURATION_REGEX = "cacheDuration=\"\\w*\"";
+  public static final String ISO_ZERO_SECONDS = "PT0S";
+  public static final Instant THE_FUTURE = Instant.now().plusSeconds((long) 1e8);
+  public static final String ISO_ONE_YEAR = "P1Y";
+  public static final Instant THE_PAST = Instant.now().minusSeconds(1);
+
+  String entityXml;
+  IdpMetadata metadata;
+
+  @Before
+  public void setup() throws IOException {
+    metadata = new IdpMetadata();
+    entityXml = IOUtils.toString(getClass().getResourceAsStream("/entityDescriptor.xml"), "UTF-8");
+  }
+
+  @Test
+  public void beforeEntityIsCreated() {
+    assertThat("Test uninitialized state", metadata.isMetadataExpired(), is(false));
```

Comment:
❓ If this test were to run after the other tests, would it have a different result?

Reply:
The state is reset in the `setup` method